exercise/staff_views.py

```diff
@@ -64,7 +67,25 @@ def get_common_objects(self):
         self.all = self.request.GET.get('all', None)
         self.all_url = self.exercise.get_submission_list_url() + "?all=yes"
         self.submissions = qs if self.all else qs[:self.default_limit]
-        self.note("all", "all_url", "submissions", "default_limit")
+        self.regrade_progress = ""
+        
+        # For the time being only admins can start mass regrade.
+        # This condition may be loosened later on
+        if self.request.user.is_superuser:
+            self.allow_regrade = True
+        else:
+            self.allow_regrade = False
+
+        if self.exercise.regrade_task:
+            result = AsyncResult(self.exercise.regrade_task)
+            if result.state == 'FAILURE':
```

Comment:
Does Celery have any constants for these constant-looking values `"FAILURE"` or `"PROGRESS"`?

Reply:
It seems no, at least all examples I could find use strings.